### ISSUE TITLE
[Layout/ExtraSpacing] Using emojis in String changes behaviour

### ISSUE THREAD

**Issue by pil0u** [rubocop/rubocop#10961]
Given this object:
```ruby
toto = [
  ["Abc",          1],
  ["Bcdef",        10],
  ["Cdef",         100],
  ["Defghijklm",   100],
  ["Efg ⭐",       1000],
  ["Fghij",        1000],
  ["Ghijklmnopqr", 10]
]
```
I choose to align the numbers using spaces relative to the longest String in the Array, in that case the last one.

I would expect this syntax to be valid, but RuboCop auto-corrects the row containing an emoji like this:
```ruby
toto = [
  ["Abc",          1],
  ["Bcdef",        10],
  ["Cdef",         100],
  ["Defghijklm",   100],
  ["Efg ⭐", 1000],
  ["Fghij",        1000],
  ["Ghijklmnopqr", 10]
]
```

If I remove the emoji, the initial syntax is considered valid and is not corrected:
```ruby
toto = [
  ["Abc",          1],
  ["Bcdef",        10],
  ["Cdef",         100],
  ["Defghijklm",   100],
  ["Efg",          1000],
  ["Fghij",        1000],
  ["Ghijklmnopqr", 10]
]
```

## RuboCop version

```
$ bundle exec rubocop -V
1.35.1 (using Parser 3.1.2.1, rubocop-ast 1.21.0, running on ruby 3.1.2 arm64-darwin21)
  - rubocop-performance 1.14.3
  - rubocop-rails 2.15.2
```

With this exact configuration:
```yml
require:
  - rubocop-performance
  - rubocop-rails

AllCops:
  NewCops: enable

Layout/LineLength:
  Enabled: false

Lint/MissingSuper:
  Exclude:
    - "**/app/components/**/*"

Metrics/AbcSize:
  Enabled: false

Metrics/BlockLength:
  Enabled: false

Metrics/ClassLength:
  Enabled: false

Metrics/CyclomaticComplexity:
  Enabled: false

Metrics/MethodLength:
  Enabled: false

Metrics/PerceivedComplexity:
  Enabled: false

Naming/VariableNumber:
  EnforcedStyle: snake_case

Rails/I18nLocaleTexts:
  Enabled: false

Rails/BulkChangeTable:
  Enabled: false

Rails/SkipsModelValidations:
  Enabled: false

Style/Documentation:
  Enabled: false

Style/NumericPredicate:
  EnforcedStyle: comparison

Style/StringLiterals:
  EnforcedStyle: double_quotes
```

**Comment by koic** [comment]
It's difficult to tell whether this is a bug or not. In my environment it looked like one space was missing before `1000`. And there is no warning when spaces are added.

```diff
 toto = [
   ["Abc",          1],
   ["Bcdef",        10],
   ["Cdef",         100],
   ["Defghijklm",   100],
-  ["Efg ⭐",       1000],
+  ["Efg ⭐",        1000],
   ["Fghij",        1000],
   ["Ghijklmnopqr", 10]
 ]
```

Below are the before and after screen captures.

Before:

<img width="193" alt="before" src="https://user-images.githubusercontent.com/13203/186637908-da875516-b303-4afe-8701-b708e61b4504.png">

After:

<img width="196" alt="after" src="https://user-images.githubusercontent.com/13203/186637983-0a1a563a-441f-41ca-a8fe-600e61301002.png">

**Comment by pil0u** [comment]
Your screenshots are really helpful.

This is a tricky one: emojis are displayed across 2 characters in my editor, but they account for 1 only. So the display I get is the same as the one you can see in my initial message on GitHub:
```ruby
toto = [
  ["Abc",          1],
  ["Bcdef",        10],
  ["Cdef",         100],
  ["Defghijklm",   100],
  ["Efg ⭐",       1000],
  ["Fghij",        1000],
  ["Ghijklmnopqr", 10]
]
```

But the truth lies in your screenshot: a space was definitely missing for RuboCop.
```ruby
toto = [
  ["Abc",          1],
  ["Bcdef",        10],
  ["Cdef",         100],
  ["Defghijklm",   100],
  ["Efg ⭐",        1000],
  ["Fghij",        1000],
  ["Ghijklmnopqr", 10]
]
```
is the correct approach.

I could have searched this one for a long time, now I am curious if I can render that just like you in VSCode.

Thank you for your help, I am closing the issue.
